text/0000-match-ergonomics.md

```diff
@@ -3,11 +3,15 @@
 - RFC PR: (leave this empty)
 - Rust Issue: (leave this empty)
 
+I'm writing comments inline (not making changes), hopefully they'll stand out in the diff.
+
 # Summary
 [summary]: #summary
 
 Better ergonomics for pattern-matching on references.
 
+AIUI, this will only apply to match and similar expressions, not pattern matching in general?
```

Comment:
Could you add this to the RFC please? For part 3, I meant when patterns are `|`ed together, I don't think there is an example of that atm. I expect that they would be treated as if they were separate arms.